### ISSUE TITLE
Transaction keeps failing saying "NFTokenTaxon is required but is missing" although the field is set.

### ISSUE THREAD

**Issue by niraj-khatiwada** [XRPL-Labs/XUMM-Issue-Tracker#398]
For the transaction type "NFTokenMint", the sign always throws "NFTokenTaxon is required but is missing" error even though the payload has the field:
```
 async getNFTMintPayload(options = DEFAULT_OPTIONS.getNFTMintPayload) {
    options = { ...DEFAULT_OPTIONS.getNFTMintPayload, ...options };

    if (options.userToken == null) {
      throw new Error('userToken is required');
    }

    if (options.toWalletAddress == null) {
      throw new Error('toWalletAddress is required');
    }

    if (options.url == null) {
      throw new Error('url is required');
    }

    return sdk.payload.create(
      {
        user_token: options.userToken,
        TransactionType: 'NFTokenMint',
        txblob: xrpl.convertStringToHex(
          JSON.stringify({
            TransactionType: 'NFTokenMint',
            Account: options.toWalletAddress,
            NFTokenTaxon: 0,
            Flags: 8,
            // Fee: '10',
            URI: xrpl.convertStringToHex(options.url),
            Memos: [
              {
                Memo: {
                  MemoType: '687474703A2F2F6578616D706C652E636F6D2F6D656D6F2F67656E65726963',
                  MemoData: '72656E74',
                },
              },
            ],
          }),
        ),

        options: { submit: true, multisign: false, expire: 15, return_url: BASE_URl },
      },
      true,
    );
  }
```
`xumm-sdk: ^1.3.1`

![image](https://user-images.githubusercontent.com/51436631/172936521-35aaf68e-6eb2-46db-be06-95ee6bad2e15.png)

What am I doing wrong?

**Comment by niraj-khatiwada** [comment]
> @niraj-khatiwada I just replayed your exact payload and signed just fine.
> 
> Can you check a couple of things:
> 
> * Are you on the latest Xumm version? 2.3.0?
> * Can you try by sending JSON to our platform, instead of binary? So the JSON (not stringified) in `txjson`?

I am on the XUMM Latest IOS app 2.3.0. I am using the xumm-sdk: 1.3.1 on backend.

If I send the txjson, the create payload actually throws this error. That's the reason why I am using txblob.

Error Message:
![image](https://user-images.githubusercontent.com/51436631/173018952-d955f3cd-89fa-49f3-b3e8-b02853629a15.png)

Using txjson:
![image](https://user-images.githubusercontent.com/51436631/173019087-6776d08e-ce7d-43e2-8c19-eb7d466cf913.png)

And one weird thing, why does the payload resolves when I pass json in txblob? Isn't it supposed to be string? The below payload format also works for some reason.
![image](https://user-images.githubusercontent.com/51436631/173019393-fa4c5d15-e00c-4bcb-a0d5-928f0f6c6ab3.png)

**Comment by N3TC4T** [comment]
@niraj-khatiwada As the error massage says The `NFTokenTaxon` is required and cannot be zero.

**Comment by WietseWind** [comment]
The reason the `txjson` doesn't work means there already is a problem to begin with. By encoding to HEX yourself and delivering a `txblob` you're just moving the problem.

Please note that passing JSON to the `txblob` does not do anything, of course that doesn't throw the initial error because it's not even processing it.

**Comment by niraj-khatiwada** [comment]
> The reason the `txjson` doesn't work means there already is a problem to begin with. By encoding to HEX yourself and delivering a `txblob` you're just moving the problem.
> 
> Please note that passing JSON to the `txblob` does not do anything, of course that doesn't throw the initial error because it's not even processing it.
> 
> I'd love to see the raw JSON you're sending.

Here is the raw json payload:

```
{
  user_token: '69002d91-8692-4ab2-8e7f-51d0a7d36082',
  TransactionType: 'NFTokenMint',
  txblob: {
    TransactionType: 'NFTokenMint',
    Account: 'rpg6edjv1pdn2s2xbo93kqhrhkgixmybeq',
    NFTokenTaxon: 0,
    Flags: 8,
    URI: '697066732E6D6F72616C69732E696F',
    Memos: [
      {
        Memo: {
          MemoType: '687474703A2F2F6578616D706C652E636F6D2F6D656D6F2F67656E65726963',
          MemoData: '72656E74'
        }
      }
    ]
  },
  options: {
    submit: true,
    multisign: false,
    expire: 15,
    return_url: 'http://localhost:3001'
  }
}
```

**Comment by WietseWind** [comment]
It makes perfect sense that transaction is rejected: the XRPL account address is invalid:
`rpg6edjv1pdn2s2xbo93kqhrhkgixmybeq`

It looks to me that it's fully lowercase, while XRPL account addresses are case sensitive.

**Comment by niraj-khatiwada** [comment]
> It makes perfect sense that transaction is rejected: the XRPL account address is invalid: `rpg6edjv1pdn2s2xbo93kqhrhkgixmybeq`
> 
> It looks to me that it's fully lowercase, while XRPL account addresses are case sensitive.
> 
> The reason that the `txblob` is completely malformed is that you:
> 
> 1. Take the JSON
> 2. Stringify
> 3. Convert string to HEX
> 
> That's not how a `txblob` is composed: a `txblob` is a binary, deterministic encoded transaction. It should be crafted with the Binary Codec. So that explains why that isn't working at all ;)
> 
> TL;DR:
> 
> 1. Go with txjson
> 2. Use a valid account address
> 
> ... and you'll be good.

Okay got it, I didn't realized about the case sensitiveness. I have another error though. I have posted [here](https://xumm.readme.io/discuss/62a31ef0c37d800107c1b289) though as you suggested earlier.
